jctools-core/src/main/java/org/jctools/queues/BaseMpscLinkedArrayQueue.java

```diff
@@ -13,26 +13,60 @@
  */
 package org.jctools.queues;
 
-import org.jctools.util.Pow2;
-import org.jctools.util.RangeUtil;
+import static org.jctools.queues.CircularArrayOffsetCalculator.allocate;
+import static org.jctools.queues.LinkedArrayQueueUtil.length;
+import static org.jctools.queues.LinkedArrayQueueUtil.modifiedCalcElementOffset;
+import static org.jctools.util.UnsafeAccess.UNSAFE;
+import static org.jctools.util.UnsafeRefArrayAccess.lvElement;
+import static org.jctools.util.UnsafeRefArrayAccess.soElement;
 
 import java.lang.reflect.Field;
 import java.util.AbstractQueue;
 import java.util.Iterator;
 
-import static org.jctools.queues.CircularArrayOffsetCalculator.allocate;
-import static org.jctools.util.UnsafeAccess.UNSAFE;
-import static org.jctools.util.UnsafeRefArrayAccess.*;
+import org.jctools.queues.IndexedQueueSizeUtil.IndexedQueue;
+import org.jctools.util.Pow2;
+import org.jctools.util.RangeUtil;
 
-abstract class BaseMpscLinkedArrayQueuePad1<E> extends AbstractQueue<E>
+abstract class BaseMpscLinkedArrayQueuePad1<E> extends AbstractQueue<E> implements IndexedQueue
 {
     long p01, p02, p03, p04, p05, p06, p07;
     long p10, p11, p12, p13, p14, p15, p16, p17;
 }
 
+// $gen:ordered-fields
 abstract class BaseMpscLinkedArrayQueueProducerFields<E> extends BaseMpscLinkedArrayQueuePad1<E>
 {
+    private final static long P_INDEX_OFFSET;
+    
+    static {
+        try
+        {
+            Field iField = BaseMpscLinkedArrayQueueProducerFields.class.getDeclaredField("producerIndex");
+            P_INDEX_OFFSET = UNSAFE.objectFieldOffset(iField);
+        }
+        catch (NoSuchFieldException e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
     protected long producerIndex;
+
+    @Override
+    public final long lvProducerIndex()
```

Comment:
These accessor methods were promoted to the parent padding class to be consistent with the array queue implementations